src/labelle/lib/render_engines/horizontally_combined.py

```diff
@@ -27,17 +29,24 @@ def render(self, context: RenderContext) -> Image.Image:
             merged_bitmap = bitmaps[0]
         else:
             label_height = max(b.height for b in bitmaps)
-            merged_bitmap = Image.new(
-                "1",
-                (
-                    sum(b.width for b in bitmaps) + self.PADDING * (len(bitmaps) - 1),
-                    label_height,
-                ),
-            )
+            total_width = _compute_total_width(bitmaps, self.padding)
+            merged_bitmap = Image.new("1", (total_width, label_height))
             x_offset = 0
             for bitmap in bitmaps:
-                y_offset = (label_height - bitmap.size[1]) // 2
+                y_offset = (label_height - bitmap.height) // 2
                 merged_bitmap.paste(bitmap, box=(x_offset, y_offset))
-                x_offset += bitmap.width + self.PADDING
+                x_offset += bitmap.width + self.padding
 
         return merged_bitmap
+
+
+def _compute_total_width(bitmaps: Sequence[Image.Image], padding: int) -> int:
+    number_of_bitmaps = len(bitmaps)
+    width_of_bitmaps = sum(b.width for b in bitmaps)
+
+    # We alternate bitmaps and padding, and have bitmaps on each end.
+    number_of_paddings = number_of_bitmaps - 1
+    width_of_paddings = number_of_paddings * padding
+
+    total_width = width_of_bitmaps + width_of_paddings
+    return total_width
```

Comment:
I am not sure this decomposition of the equation to calculate total width is neccessary. It is too detailed.
You could just replace `self.PADDING` with `padding` in the original code.
I am fond of the idea to create intermediate `total_width` variable, since it helps readability, but to extract a one-liner to a function seems to me as an exaggaration.
Also, if you do extrac to a separate function, why not make is a part of the class?

Reply:
I made it verbose because I don't ever want to make an effort to reconstruct the reason why I did `- 1`. Also, splitting it off as a function makes it easy to ignore the uninteresting detail, even if it could be done as a one-liner.

I don't see what the width computation has to do with the `RenderEngine` interface, which is why I didn't add it to the class.